### PR TITLE
Initial implementation of materialization, materializes top level que…

### PR DESCRIPTION
**Pull request by nachoarreola**
…ries.

### CODE REVIEW

**Comment by nachoarreola** [comment]
@christopherswenson thanks for taking a look at this!

**Comment by christopherswenson** [comment]
@nachoarreola Awesome work.

File: packages/malloy/src/malloy.ts [hunk 1]
```diff
@@ -394,7 +400,11 @@ export class Malloy {
           }
           queryModel = new QueryModel(partialModel);
         }
-        const compiledSql = queryModel.compileQuery(segment, false).sql;
+        const compiledSql = queryModel.compileQuery(
+          segment,
+          false,
```
Comment (whscullin) [inline packages/malloy/src/malloy.ts:405]: `undefined`

File: packages/malloy/src/malloy.ts [hunk 2]
```diff
@@ -879,8 +898,16 @@ export class PreparedQuery implements Taggable {
...
   public get preparedResult(): PreparedResult {
+    return this.getPreparedResult();
+  }
+
+  public getPreparedResult(options?: PrepareResultOptions): PreparedResult {
     const queryModel = new QueryModel(this._modelDef);
-    const translatedQuery = queryModel.compileQuery(this._query);
+    const translatedQuery = queryModel.compileQuery(
+      this._query,
+      false,
```
Comment (whscullin) [inline packages/malloy/src/malloy.ts:908]: `undefined`

File: packages/malloy/src/malloy.ts (hunk 1 above)
Comment (whscullin) [reply packages/malloy/src/malloy.ts]: psych! should be `false`

**Pull request merged**